### ISSUE TITLE
 Issues with compiling the databases

### ISSUE THREAD

**Issue by kezia-drane** [arumugamlab/MIntO#7]
what's failing is the eggnog_db - log saying the py package created by the database creator is failing and cannot find its related files?

I also tried the manual instructions.  In the instructions for eggnog_db it says
"Create a conda environment that will be used to download the eggNOG database conda create env $MINTO_DIR/envs/py38_env.yml --name py38"
but there is no py38_env.yml in the envs directory?

Thank you in advance.

**Comment by microbiomix** [comment]
Hello,

Manual installation of databases does not work anymore and we forgot to remove that from the instructions. My apologies.

Please try the automatic installation of dependencies and post the detailed error message here and we can try to help you.

**Comment by kezia-drane** [comment]
Hi Mani,

I've attached the error message that we are getting, any additional help would be great.

Thank you,
Kezia

Kezia Drane
PhD Candidate | James Cook University
College of Public Health, Medical and Veterinary Sciences
Townsville 4811, QLD, Australia
Ph: 04 09497544
Email: ***@***.******@***.***>

**Comment by arumugamlab** [comment]
Hi Kezia,
I cannot see the attachment. If you attach the file replying to the comment at GitHub, then I can see it.
Best,
Mani

**Comment by kezia-drane** [comment]
Hi Mani,
Please see error below:

For info I'm using
Ubuntu 22.04
Miniconda 3 latest
[https://github.com/arumugamlab/MIntO/archive/refs/tags/1.0.1.tar.gz](https://aus01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Farumugamlab%2FMIntO%2Farchive%2Frefs%2Ftags%2F1.0.1.tar.gz&data=05%7C01%7Ckezia.drane%40my.jcu.edu.au%7C9f71ef199e52434b229f08db5119a8cf%7C2eba4cf8af764db3bcaf81b5592535ef%7C0%7C0%7C638192943950559442%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=8NZiMiBRpq4q8fXHifuQO0W%2Fx53tdxjh7wLEtaQi0As%3D&reserved=0)
 

error message:

+ snakemake --snakefile /opt/MIntO/smk/dependencies.smk --use-conda --restart-times 10 --keep-going --latency-wait 10 --jobs 6 --configfile /opt/MIntO/configuration/dependencies.yaml --conda-prefix /opt/MIntO/conda_env --cores 6
*******************************
Reading configuration yaml file: 
 *******************************
  
Building DAG of jobs...
Your conda installation is not configured to use strict channel priorities. This is however crucial for having robust and correct environments (for details, see [https://conda-forge.org/docs/user/tipsandtricks.html](https://aus01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fconda-forge.org%2Fdocs%2Fuser%2Ftipsandtricks.html&data=05%7C01%7Ckezia.drane%40my.jcu.edu.au%7C9f71ef199e52434b229f08db5119a8cf%7C2eba4cf8af764db3bcaf81b5592535ef%7C0%7C0%7C638192943950559442%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=sqifE2kJAF9gAPDcuaXRB%2BL0nj6mdt30xGh8uwa%2B7K8%3D&reserved=0)). Please consider to configure strict priorities by executing 'conda config --set channel_priority strict'.
Creating conda environment MIntO/envs/MIntO_base.yml...
Downloading and installing remote packages.
Environment for /opt/MIntO/envs/MIntO_base.yml created (location: MIntO/conda_env/806043aae1e8b1e0bb5fb2a7193dcad3_)
Creating conda environment MIntO/envs/checkm.yaml...
Downloading and installing remote packages.
Environment for /opt/MIntO/envs/checkm.yaml created (location: MIntO/conda_env/d7a5614dd0902c880fa5336aae2b76d7_)
Creating conda environment MIntO/envs/phylophlan3.yaml...
Downloading and installing remote packages.
Environment for /opt/MIntO/envs/phylophlan3.yaml created (location: MIntO/conda_env/f9403b6a6f599c3a0830dd7ce399c224_)
Creating conda environment MIntO/envs/py36_env.yml...
Downloading and installing remote packages.
Environment for /opt/MIntO/envs/py36_env.yml created (location: MIntO/conda_env/3b6f8154b346e1a7b8349ce52b7f642d_)
Creating conda environment MIntO/envs/dbcan_env.yml...
Downloading and installing remote packages.
Environment for /opt/MIntO/envs/dbcan_env.yml created (location: MIntO/conda_env/b45b50e73d05899c7cb01e24ddf2fab3_)
Creating conda environment MIntO/envs/motus_env.yml...
Downloading and installing remote packages.
CreateCondaEnvironmentException:
Could not create conda environment from /opt/MIntO/envs/motus_env.yml:
Command:
mamba env create --quiet --file "/opt/MIntO/conda_env/f15023b2b930d1499769dca509ccae32_.yaml" --prefix "/opt/MIntO/conda_env/f15023b2b930d1499769dca509ccae32_"
Output:
Could not solve for environment specs
The following packages are incompatible
├─ curl ==7.80.0 h494985f_1 is requested and can be installed;
├─ htslib ==1.9 h4da6232_3 is requested and can be installed;
├─ libdeflate ==1.2 h516909a_1 is requested and can be installed;
├─ libgcc-ng ==11.2.0 h1d223b6_11 is requested and can be installed;
├─ libzlib ==1.2.11 h36c2ea0_1013 is installable and it requires
│  └─ zlib 1.2.11 *_1013, which can be installed;
├─ motus ==2.1.1 py37_3 is installable and it requires
│  └─ samtools   with the potential options
│     ├─ samtools [0.1.16|0.1.17|...|1.7] would require
│     │  └─ openssl >=1.1.0,<=1.1.1  with the potential options
│     │     ├─ openssl [1.1.1a|1.1.1b|...|1.1.1t], which can be installed;
│     │     └─ openssl [1.1.1p|1.1.1q|...|3.1.0] would require
│     │        └─ libgcc-ng >=12 , which conflicts with any installable versions previously reported;
│     ├─ samtools [0.1.12|0.1.13|...|1.7] would require
│     │  └─ zlib [1.2.8 |1.2.8* ], which conflicts with any installable versions previously reported;
│     ├─ samtools [0.1.12|0.1.13|...|1.9] would require
│     │  └─ ncurses [5.9* |>=5.9,<5.10.0a0 ], which can be installed;
│     ├─ samtools [0.1.18|0.1.19|1.1] would require
│     │  └─ ncurses >=6.3,<7.0a0 , which can be installed;
│     ├─ samtools [0.1.19|1.1|1.9] would require
│     │  └─ ncurses >=6.1,<6.2.0a0 , which can be installed;
│     ├─ samtools 1.10 would require
│     │  └─ htslib >=1.10.2,<1.11.0a0 , which conflicts with any installable versions previously reported;
│     ├─ samtools 1.10 would require
│     │  └─ htslib >=1.10,<1.11.0a0 , which conflicts with any installable versions previously reported;
│     ├─ samtools 1.11 would require
│     │  └─ htslib >=1.11,<1.12.0a0 , which conflicts with any installable versions previously reported;
│     ├─ samtools 1.12 would require
│     │  └─ htslib >=1.12,<1.13.0a0 , which conflicts with any installable versions previously reported;
│     ├─ samtools 1.13 would require
│     │  └─ htslib >=1.13,<1.14.0a0 , which conflicts with any installable versions previously reported;
│     ├─ samtools [1.14|1.15] would require
│     │  └─ htslib >=1.14,<1.15.0a0 , which conflicts with any installable versions previously reported;
│     ├─ samtools [1.15|1.15.1] would require
│     │  └─ htslib >=1.15,<1.16.0a0 , which conflicts with any installable versions previously reported;
│     ├─ samtools [1.15.1|1.16.1] would require
│     │  └─ htslib >=1.16,<1.17.0a0 , which conflicts with any installable versions previously reported;
│     ├─ samtools [1.16.1|1.17] would require
│     │  └─ htslib >=1.17,<1.18.0a0 , which conflicts with any installable versions previously reported;
│     ├─ samtools [1.3|1.3.1|1.6] would require
│     │  └─ curl >=7.81.0,<8.0a0  but there are no viable options
│     │     ├─ curl [7.81.0|7.82.0|...|7.88.1] conflicts with any installable versions previously reported;
│     │     └─ curl 7.83.0 would require
│     │        └─ libcurl 7.83.0 h0b77cf5_0, which does not exist (perhaps a missing channel);
│     └─ samtools 1.9 would require
│        └─ libdeflate >=1.0,<1.1.0a0 , which conflicts with any installable versions previously reported;
├─ ncurses ==6.2 h58526e2_4 is uninstallable because it conflicts with any installable versions previously reported;
├─ openssl ==3.0.0 h7f98852_2 is uninstallable because it conflicts with any installable versions previously reported;
├─ python ==3.7.12 hf930737_100_cpython is uninstallable because it requires
│  └─ openssl >=3.0.0,<4.0a0  but there are no viable options
│     ├─ openssl [1.1.1p|1.1.1q|...|3.1.0], which cannot be installed (as previously explained);
│     ├─ openssl 3.0.0 conflicts with any installable versions previously reported;
│     └─ openssl [3.0.0|3.0.2|3.0.3] conflicts with any installable versions previously reported;
└─ samtools ==1.7 1, which can be installed (as previously explained).

**Comment by arumugamlab** [comment]
Hi,

Apologies for the delay. I see that you have used version 1.0.1. We have just released v2.0.0-beta.1. Things should run smoothly from installation until final analysis. We have also included a test script [here](https://github.com/arumugamlab/MIntO/blob/main/testing/run_IBD_test.sh). Please clone this version using:

```
git clone https://github.com/arumugamlab/MIntO.git
git checkout tags/2.0.0-beta.1
```
and let me know if you still run into issues installing dependencies.

Best wishes,
Mani

**Comment by arumugamlab** [comment]
Closing as there was no feedback or follow up. Please reopen if it continues to be an issue.
